compiler/noirc_frontend/src/tests/unused_items.rs

```diff
@@ -157,3 +160,14 @@ fn errors_on_unused_trait() {
     assert_eq!(ident.to_string(), "Foo");
     assert_eq!(*item_type, "trait");
 }
+
+#[test]
```

Comment:
Please add tests for the struct/trait cases

Reply:
What should be these cases? This PR only handles local variables.

Reply:
Checking that attributes on structs/traits throw errors/warnings

Reply:
Oh, you mean, if you use `allow(unused_variables)` on a struct or a trait then it should give an error?

Right now we don't error on these cases. For example you can say `#[deprecated] struct Foo {}` and it's completely ignored.

I'm not against adding those checks, I'm just not sure we should do that in this PR.